dbms/src/Common/HashTable/StringHashTable.h

```diff
@@ -0,0 +1,494 @@
+#pragma once
+
+#include <Common/HashTable/HashMap.h>
+#include <Common/HashTable/HashTable.h>
+
+/// TODO feature macros
+
+#define CASE_1_8 \
+    case 1: \
+    case 2: \
+    case 3: \
+    case 4: \
+    case 5: \
+    case 6: \
+    case 7: \
+    case 8
+
+#define CASE_9_16 \
+    case 9: \
+    case 10: \
+    case 11: \
+    case 12: \
+    case 13: \
+    case 14: \
+    case 15: \
+    case 16
+
+#define CASE_17_24 \
+    case 17: \
+    case 18: \
+    case 19: \
+    case 20: \
+    case 21: \
+    case 22: \
+    case 23: \
+    case 24
+
+struct StringKey0
+{
+};
+
+using StringKey8 = UInt64;
+using StringKey16 = DB::UInt128;
+struct StringKey24
+{
+#if !__clang__
+#    pragma GCC diagnostic push
+#    pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+#endif
```

Comment:
Which warning does this hide? I'm not seeing any with either GCC 7.4 or 8.3. Also this should probably use IGNORE_MAYBE_UNINITIALIZED_PUSH, like ColumnsHashingImpl.h

Reply:
Ah, that's from dbms/src/Common/UInt128.h . I'm not sure if that pragma is needed either. Hopefully we can get rid of that.